src/cli/tree.rs

```diff
@@ -0,0 +1,396 @@
+            ))?;
+        }
+    }
+
+    let mut visited_pkgs = Vec::new();
+    let direct_dep_count = direct_deps.len();
+
+    for (index, pkg_name) in direct_deps.iter().enumerate() {
+        visited_pkgs.push(pkg_name.to_owned());
+
+        let last = index == direct_dep_count - 1;
+        let symbol = if last {
+            UTF8_SYMBOLS.ell
+        } else {
+            UTF8_SYMBOLS.tee
+        };
+        if let Some(pkg) = dep_map.get(pkg_name) {
+            println!(
+                "{} {} v{}",
+                symbol,
+                console::style(pkg.name.clone()).fg(Color::Green).bold(),
+                pkg.version
+            );
+
+            let prefix = if last {
+                UTF8_SYMBOLS.empty
+            } else {
+                UTF8_SYMBOLS.down
+            };
+            print_dependency_leaf(pkg, format!("{} ", prefix), dep_map, &mut visited_pkgs)
+        }
+    }
+    Ok(())
+}
+
+/// Recursively print top down dependency tree nodes
+fn print_dependency_leaf(
+    pkg: &Pkg,
+    prefix: String,
+    dep_map: &HashMap<String, Pkg>,
+    visited_pkgs: &mut Vec<String>,
+) {
+    let dep_count = pkg.dependencies.len();
+    for (index, dep_name) in pkg.dependencies.iter().enumerate() {
+        let last = index == dep_count - 1;
+        let symbol = if last {
+            UTF8_SYMBOLS.ell
+        } else {
+            UTF8_SYMBOLS.tee
+        };
+
+        // skip virtual packages
+        if dep_name.starts_with("__") {
+            continue;
+        }
+
+        if let Some(dep) = dep_map.get(dep_name) {
+            let visited = visited_pkgs.contains(&dep.name);
+            visited_pkgs.push(dep.name.to_owned());
+
+            println!(
+                "{}{} {} v{} {}",
+                prefix,
+                symbol,
+                dep.name,
+                dep.version,
+                if visited { "(*)" } else { "" }
+            );
+
+            if visited {
+                continue;
+            }
+
+            let new_prefix = if last {
+                format!("{}{} ", prefix, UTF8_SYMBOLS.empty)
+            } else {
+                format!("{}{} ", prefix, UTF8_SYMBOLS.down)
+            };
+            print_dependency_leaf(dep, new_prefix, dep_map, visited_pkgs);
+        }
+    }
+}
+
+/// Extract the direct Conda and PyPI dependencies from the environment
+fn direct_dependencies(
+    environment: &crate::project::Environment<'_>,
+    platform: &Platform,
+) -> Vec<String> {
+    let mut project_dependency_names = environment
+        .dependencies(None, Some(*platform))
+        .names()
+        .map(|p| p.as_source().to_string())
+        .collect_vec();
+    project_dependency_names.extend(
+        environment
+            .pypi_dependencies(Some(*platform))
+            .into_iter()
+            .map(|(name, _)| name.as_normalized().to_string()),
+    );
+    project_dependency_names
+}
+
+#[derive(Debug)]
+struct Pkg {
```

Comment:
We generally like to use full names in the structs. So possibly `PackageLeaf` or `SimplePackage`. I'm also not best at naming but Pkg seems a bit to un-descriptive on first read.

Reply:
I may have gone a little overboard shortening things and trying to avoid collisions with existing names. What about `PackageNode`?

Reply:
Also makes sense! 

Note, (possibly not the right choice here but), you can skip the `use rattler_lock::Package;` and then you can just use `rattler_lock::Package` directly, especially as it is only used once.  Sometimes this is helpful.